src/System.IO.FileSystem/tests/Directory/GetFiles_str_str_so.cs

```diff
@@ -8,642 +8,321 @@
 
-            //Scenario 7: dir is readonly
-            try
+            for (int i = 0; i < files.Length; i++)
             {
-                //Readonly
-                dirName = ManageFileSystem.GetNonExistingDir(Directory.GetCurrentDirectory(), ManageFileSystem.DirPrefixName);
-                using (ManageFileSystem fileManager = new ManageFileSystem(dirName, 3, 100))
-                {
-                    //now lets make this directory readonly?
-                    new DirectoryInfo(dirName).Attributes = new DirectoryInfo(dirName).Attributes | FileAttributes.ReadOnly;
-
-                    expectedFiles = fileManager.GetAllFiles();
-                    list = new List<String>(expectedFiles);
-                    files = Directory.GetFiles(dirName, "*.*", SearchOption.AllDirectories);
-                    Eval(files.Length, list.Count, "Err_862vhr! wrong file count.");
-
-                    for (int i = 0; i < files.Length; i++)
-                    {
-                        if (Eval(list.Contains(files[i]), "Err_556ioj! unexpected file found: {0}", files[i]))
-                            list.Remove(files[i]);
-                    }
-                    if (!Eval(list.Count == 0, "Err_562xwh! {0} expected files not found.", list.Count))
-                    {
-                        foreach (String fileName in list)
-                            Console.WriteLine(fileName);
-                    }
-                    if (Directory.Exists(dirName) && (new DirectoryInfo(dirName).Attributes & FileAttributes.ReadOnly) != 0)
-                        new DirectoryInfo(dirName).Attributes = new DirectoryInfo(dirName).Attributes ^ FileAttributes.ReadOnly;
-                }
+                if (Eval(list.Contains(files[i]), "Err_556ioj! unexpected file found: {0}", files[i]))
+                    list.Remove(files[i]);
             }
-            catch (Exception ex)
+            if (!Eval(list.Count == 0, "Err_562xwh! {0} expected files not found.", list.Count))
             {
-                s_pass = false;
-                Console.WriteLine("Err_418qfv! Exception caught in scenario: {0}", ex);
+                foreach (String fileName in list)
+                    Console.WriteLine(fileName);
             }
+            if (Directory.Exists(dirName) && (new DirectoryInfo(dirName).Attributes & FileAttributes.ReadOnly) != 0)
+                new DirectoryInfo(dirName).Attributes = new DirectoryInfo(dirName).Attributes ^ FileAttributes.ReadOnly;
         }
-        catch (Exception ex)
-        {
-            s_pass = false;
-            Console.WriteLine("Err_234rsgf! Uncaught exception in RunTest: {0}", ex);
-        }
-
-        Assert.True(s_pass);
     }
 
+    #endregion
+
     //Checks for error
-    private static bool Eval(bool expression, String msg, params Object[] values)
+    protected static bool Eval(bool expression, String msg, params Object[] values)
```

Comment:
Are these Eval methods just legacy from before we had Assert.Equal?  If so, could we remove them and just use Assert directly?